packages/core/src/types.ts

```diff
@@ -73,6 +77,23 @@ export type Action<TContext, TEvent extends EventObject> =
   | ActionObject<TContext, TEvent>
   | ActionFunction<TContext, TEvent>;
 
+export type BaseAction<
+  TContext,
+  TEvent extends EventObject,
+  TAction extends BaseActionObject
+> =
+  | TAction['type']
+  | TAction
+  | SendAction<TContext, TEvent, any>
+  | AssignAction<TContext, TEvent>
+  | ActionFunction<TContext, TEvent>;
```

Comment:
@Andarist This makes types pass but I feel like we've had it before and removed it for some reason (I could be wrong), do you think this is fine?

Reply:
Hard to tell without giving this a spin locally - I think though that the list is missing some other builtin actions like cancel, log etc.